.github/workflows/runtime-image-update.yaml

```diff
@@ -0,0 +1,105 @@
+---
+name: Update runtime image tag
+
+on:  # yamllint disable-line rule:truthy
+  schedule:
+    - cron: '0 0 * * *'  # Run every day at midnight UTC
+  workflow_dispatch:
+    inputs:
+      quay_repo:
+        description: 'Quay Repository'
+        default: 'astronomer/astro-runtime'
+
+jobs:
+  runtime-image-tag-update-job:
+    runs-on: 'ubuntu-20.04'
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+
+      - name: Install dependencies
+        run: pip install requests semantic-version
+
+      - name: Get current tag from Dockerfile.astro_cloud
+        id: get_current_tag
+        # yamllint disable rule:line-length
+        run: |
+          CURRENT_TAG=$(awk -F: '/quay.io\/astronomer\/astro-runtime/ {print $NF}' ./.circleci/integration-tests/Dockerfile.astro_cloud)
+          echo "CURRENT_TAG=$CURRENT_TAG"
+          echo "::set-output name=tag::$CURRENT_TAG"
+        # yamllint enable rule:line-length
+
+      - name: Get latest tag from Quay.io
+        id: get_latest_tag
+        # yamllint disable rule:line-length
+        run: |
+          LATEST_TAG=$(python ./.github/scripts/get_latest_runtime_image_tag.py ${{ github.event.inputs.quay_repo }})
+          echo "LATEST_TAG=$LATEST_TAG"
+          echo "::set-output name=tag::$LATEST_TAG"
+        # yamllint enable rule:line-length
+
+      - name: Check if tag was updated
+        id: check_tag_updated
+        run: |
+          if [ "${{ steps.get_current_tag.outputs.tag }}" != "${{ steps.get_latest_tag.outputs.tag }}" ]; then
+            echo "Tag was updated"
+            echo "::set-output name=tag_updated::true"
+          else
+            echo "Tag was not updated"
+            echo "::set-output name=tag_updated::false"
+          fi
+
+      - name: Create runtime image tag update branch
+        id: create_runtime_image_update_branch
+        run: |
+          BRANCH_NAME="update-runtime-image-tag-${{ steps.get_latest_tag.outputs.tag }}"
+          git fetch origin main
+          git checkout main
```

Comment:
I'm not sure whether it's necessary. I thought we've checkout to main through https://github.com/astronomer/astronomer-providers/pull/1159/files#diff-09763d6cc66b999842e6f6520c49b659d5d8e89185fb9b64704e07c69f473e42R18-R19

Reply:
The Github actions job was failing in my PR without this. Only after introducing this the Actions job finally ran fine.